pkg/cloud/scope/cluster.go

```diff
@@ -192,14 +192,6 @@ func (s *ClusterScope) ControlPlaneLoadBalancers() []*infrav1.AWSLoadBalancerSpe
 	}
 }
 
-// ControlPlaneLoadBalancerScheme returns the Classic ELB scheme (public or internal facing).
-func (s *ClusterScope) ControlPlaneLoadBalancerScheme() infrav1.ELBScheme {
-	if s.ControlPlaneLoadBalancer() != nil && s.ControlPlaneLoadBalancer().Scheme != nil {
-		return *s.ControlPlaneLoadBalancer().Scheme
-	}
-	return infrav1.ELBSchemeInternetFacing
-}
```

Comment:
@mtulio Apologies for the back and forth on this, let's just add `// Deprecated` here instead? That should trigger the linter, and get rid of the API compatibility promise

Reply:
Updated here https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4840#discussion_r1514979642